public/app/features/dashboard-scene/settings/JsonModelEditView.tsx

```diff
@@ -53,11 +81,11 @@ export class JsonModelEditView extends SceneObjectBase<JsonModelEditViewState> i
             showLineNumbers={true}
             showMiniMap={true}
             containerStyles={styles.codeEditor}
+            onBlur={model.onCodeEditorBlur}
           />
           {canSave && (
             <div>
-              {/**TODO: add saving functionality */}
-              <Button type="submit">
+              <Button type="submit" onClick={model.onApplyChange}>
                 <Trans i18nKey="dashboard-settings.json-editor.save-button">Save changes</Trans>
```

Comment:
I don't think this should save Save changes, maybe "Apply" or "Load" , as it does not save anything (unlike in the old dashboard arch). I think it's good that it's not saving, so you can view the change before saving

Reply:
Makes sense, I'll update this

Reply:
@torkelo Actually, do we want to update this right now because it looks like we would have to update the internationalization files?

Reply:
@harisrozajac we would have to add a new text entry (so we do not change the old arch version)

Reply:
added a new text entry